internal/resource/http.go

```diff
@@ -114,38 +109,44 @@ func NewHttpClient(logger *log.Logger, fetcher Fetcher, timeouts types.Timeouts,
 			_, expectedSumString, _ := ca.Verification.HashParts()
 			expectedSum, err = hex.DecodeString(expectedSumString)
 			if err != nil {
-				logger.Crit("Error parsing verification string %q: %v", expectedSumString, err)
-				return client, err
+				f.Logger.Crit("Error parsing verification string %q: %v", expectedSumString, err)
+				return err
 			}
 		}
 
-		ca, err := fetcher.FetchToBuffer(*u, FetchOptions{
+		ca, err := f.FetchToBuffer(*u, FetchOptions{
 			Hash:        hasher,
 			ExpectedSum: expectedSum,
 		})
 		if err != nil {
-			logger.Err("Unable to fetch CA (%s): %s", u, err)
-			return client, err
+			f.Logger.Err("Unable to fetch CA (%s): %s", u, err)
+			return err
 		}
 
 		block, _ := pem.Decode(ca)
 		if block == nil {
-			logger.Err("Unable to decode CA (%s)", u)
-			return client, ErrPEMDecodeFailed
+			f.Logger.Err("Unable to decode CA (%s)", u)
+			return ErrPEMDecodeFailed
 		}
 
 		cert, err := x509.ParseCertificate(block.Bytes)
 		if err != nil {
-			logger.Err("Unable to parse CA (%s): %s", u, err)
-			return client, err
+			f.Logger.Err("Unable to parse CA (%s): %s", u, err)
+			return err
 		}
 
-		logger.Info("Adding %q to list of CAs", cert.Subject.CommonName)
+		f.Logger.Info("Adding %q to list of CAs", cert.Subject.CommonName)
 		pool.AddCert(cert)
 	}
 
-	client.client.Transport = &http.Transport{
-		ResponseHeaderTimeout: time.Duration(responseHeader) * time.Second,
+	f.client.transport.TLSClientConfig = &tls.Config{RootCAs: pool}
+	f.client.client.Transport = f.client.transport
+	return nil
+}
+
+func newHttpTransport(responseHeaderTimeout int) http.RoundTripper {
```

Comment:
This function doesn't seem to be getting called anywhere.

Reply:
Whoops, I meant to delete this. Should be removed now.